azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/utils/ReturnRequestIdHeaderInterceptor.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.resources.fluentcore.utils;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import java.io.IOException;
+
+/**
+ * An interceptor for requesting server return client-request-id in response headers.
+ * Optionally, fill-in the client-request-id, if server does not return it in response headers.
+ */
+public class ReturnRequestIdHeaderInterceptor implements Interceptor {
+
+    private static final String NAME_RETURN_CLIENT_REQUEST_ID = "x-ms-return-client-request-id";
+    private static final String NAME_CLIENT_REQUEST_ID = "x-ms-client-request-id";
+
+    private final boolean copyClientRequestIdInResponse;
+
+    /**
+     * Creates a new instance of ReturnRequestIdHeaderInterceptor.
+     * Sets "x-ms-return-client-request-id: true" in requests headers.
+     */
+    public ReturnRequestIdHeaderInterceptor() {
+        this(false);
```

Comment:
Might better default `true` or just remove this boolean.

Reply:
One reason customer need the "x-ms-client-request-id" is to cross-check with service when something went wrong. If we send "x-ms-return-client-request-id=true" but service does not return request id, it might be an indicate that service is not handling "x-ms-client-request-id".